src/pages/Facility/settings/organizations/FacilityOrganizationIndex.tsx

```diff
@@ -85,43 +92,216 @@ export default function FacilityOrganizationIndex({
       </Page>
     );
   }
+  const toggleRow = (id: string) => {
+    setExpandedRows((prev) => ({ ...prev, [id]: !prev[id] }));
+  };
+  const getChildren = (parentId: string) => {
+    return data.results.filter((org) => org.parent?.id === parentId);
+  };
+  const OrganizationRow = ({
+    org,
+    expandedRows,
+    toggleRow,
+    getChildren,
+    indent,
+  }: {
+    org: {
+      id: string;
+      name: string;
+      parent?: { id: string };
+      org_type: string;
+    };
+    expandedRows: Record<string, boolean>;
+    toggleRow: (id: string) => void;
+    getChildren: (parentId: string) => {
+      id: string;
+      name: string;
+      parent?: { id: string };
+      org_type: string;
+    }[];
+    indent: number;
+  }) => {
+    const children = getChildren(org.id);
+    const isTopLevel = !org.parent || Object.keys(org.parent).length === 0;
 
+    const toggleAllChildren = () => {
+      setExpandedRows((prevExpandedRows) => {
+        const newExpandedRows = { ...prevExpandedRows };
+        const toggleChildren = (parentId: string, expand: boolean) => {
+          getChildren(parentId).forEach((child) => {
+            newExpandedRows[child.id] = expand;
+            toggleChildren(child.id, expand);
+          });
+        };
+        const shouldExpand = !children.some(
```

Comment:
```suggestion
        const shouldExpand = !children.every(
```

If user has expanded some and then clicks expand all, it should still expand rather than collapsing.